ibis/backends/duckdb/compiler.py

```diff
@@ -187,14 +187,17 @@ def visit_ArrayZip(self, op, *, arg):
             ]
         )
         func = sge.Lambda(this=body, expressions=[i])
-        return self.f.list_apply(
+        regular = self.f.list_apply(
             self.f.range(
                 1,
                 # DuckDB Range excludes upper bound
                 self.f.greatest(*map(self.f.len, arg)) + 1,
             ),
             func,
         )
+        # if any of the input arrays in arg are NULL, the result is NULL
+        any_arg_null = reduce(lambda x, y: x | y.is_(NULL), arg, False)
```

Comment:
this `reduce` was an uniformed acceptance from a github copilot suggestion (that worked, much wow), so please verify this is a good way to accomplish this.

Reply:
> please verify this is a good way to accomplish this

lol at AI

```suggestion
        any_arg_null = reduce(lambda x, y: x | y.is_(NULL), arg, FALSE)
```

The `FALSE` may need to be imported

Reply:
`sg.or_()` actually is what we want I think